tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_invalid_values.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include "testing_common_defs.h"
+#include <addr.h>
+#include <keys.h>
+#include "host_mocks/assert.h"
+#include "mocks/conn.h"
+#include "mocks/hci_core.h"
+#include "mocks/keys_help_utils.h"
+#include "mocks/hci_core_expects.h"
+
+/* This call back will be called when bt_unpair() is called with the same parameters
+ * It can be used when a call to bt_unpair() isn't expected
+ */
+static int bt_unpair_unreachable_custom_fake(uint8_t id, const bt_addr_le_t *addr)
+{
+	ARG_UNUSED(id);
+	ARG_UNUSED(addr);
+	zassert_unreachable("Unexpected call to 'bt_unpair()' occurred");
+	return 0;
+}
+
+/* This call back will be called when bt_conn_foreach() is called with the same parameters */
+static void bt_conn_foreach_custom_fake_conn_ref_null(int type, bt_conn_foreach_cb func,
+							void *data)
+{
+	expect_assert();
+	func(NULL, data);
+}
+
+/* This call back will be called when bt_conn_foreach() is called with the same parameters */
+static void bt_conn_foreach_custom_fake_data_ref_null(int type, bt_conn_foreach_cb func,
+							void *data)
+{
+	struct bt_conn conn;
+
+	expect_assert();
+	func(&conn, NULL);
+}
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list is full
+ *
+ *  Constraints:
+ *   - Keys pool list is full
+ *   - All Keys are used with a connection
+ *   - CONFIG_BT_KEYS_OVERWRITE_OLDEST is enabled
+ *
+ *  Expected behaviour:
+ *   - An assertion is triggered and execution stops
+ */
+void test_find_key_in_use_receives_null_conn_ref(void)
+{
+	uint8_t id = BT_ADDR_ID_5;
+	bt_addr_le_t *addr = BT_ADDR_LE_5;
+
+	bt_unpair_fake.custom_fake = bt_unpair_unreachable_custom_fake;
+	bt_conn_foreach_fake.custom_fake = bt_conn_foreach_custom_fake_conn_ref_null;
+
+	bt_keys_get_addr(id, addr);
+
+	/* Should not reach this point */
+	zassert_unreachable(NULL);
+}
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list is full
+ *
+ *  Constraints:
+ *   - Keys pool list is full
+ *   - All Keys are used with a connection
+ *   - CONFIG_BT_KEYS_OVERWRITE_OLDEST is enabled
+ *
+ *  Expected behaviour:
+ *   - An assertion is triggered and execution stops
+ */
+void test_find_key_in_use_receives_null_data_ref(void)
+{
+	uint8_t id = BT_ADDR_ID_5;
+	bt_addr_le_t *addr = BT_ADDR_LE_5;
+
+	bt_unpair_fake.custom_fake = bt_unpair_unreachable_custom_fake;
+	bt_conn_foreach_fake.custom_fake = bt_conn_foreach_custom_fake_data_ref_null;
+
+	bt_keys_get_addr(id, addr);
+
+	/* Should not reach this point */
+	zassert_unreachable(NULL);
+}
+
+/*
+ *  Test invalid (NULL) BT address reference
+ *
+ *  Constraints:
+ *   - Address value is NULL
+ *
+ *  Expected behaviour:
+ *   - An assertion is raised and execution stops
+ */
+void test_null_address_reference(void)
+{
+	expect_assert();
```

Comment:
why do we have an `expect_assert()` here and not in the previous fns, that should also be raising an assert (per their docstring).

Reply:
Yes, it's a bit misleading, it even took me sometime to figure out how come the test is passing if I'm not using it.
I found out that I added it in the callback functions assigned to `bt_conn_foreach_fake.custom_fake`.

Anyway, I moved it just before calling `bt_keys_get_addr()` to make it more readable.